### PR TITLE
Also store fill id in request when withdrawn to correctly resolve other active claims

### PR DESCRIPTION
**Pull request by fredo**
There was a bug in the contracts code when a claimer had multiple claims on the same request.

If one of them was successfully withdrawn, all other claims would be resolved in favor of the claimer which could be wrong if the fill id was different.

The rule is: 

All identical claims should be resolved equally. A correct claim is only correct if the claimer is the filler and the fill id matches. If the correct filler creates a claim with a wrong fill id, the claim becomes invalid.

This must be reflected in the withdraw function: In order to achieve that we need to store all relevant information with the request upon deposit withdraw, so that future claims can be resolved correctly. 
That means if the future claim is from the same address who withdrew the deposit AND the fill id is the same as from the claim which was used when withdrawing the deposit, then the claimer is entitled to receive the claim stake. In all other cases the challenger wins.

was found and is partially needed for #645 

Please note that the `FillerInfo` object will be used in an upcoming PR in the resolution registry as well

### CODE REVIEW

File: contracts/contracts/BeamerUtils.sol [hunk 1]
```diff
@@ -2,6 +2,11 @@
 pragma solidity ^0.8.12;
 
 library BeamerUtils {
+    struct FillerInfo {
```
Comment (palango) [inline contracts/contracts/BeamerUtils.sol:5]: Should we maybe call this `FillInfo`?

File: contracts/contracts/RequestManager.sol [hunk 2]
```diff
@@ -175,7 +175,7 @@ contract RequestManager is Ownable {
         newRequest.targetTokenAddress = targetTokenAddress;
         newRequest.targetAddress = targetAddress;
         newRequest.amount = amount;
-        newRequest.depositReceiver = address(0);
+        newRequest.withdrawInfo = BeamerUtils.FillerInfo(address(0), "0x0");
```
Comment (palango) [inline contracts/contracts/RequestManager.sol:178]: I think `bytes32(0)` is clearer.

Comment (fredo) [reply contracts/contracts/RequestManager.sol:178]: Okay, I wasn't sure what'S the most obvious way.

File: contracts/contracts/BeamerUtils.sol (hunk 1 above)
Comment (fredo) [reply contracts/contracts/BeamerUtils.sol]: agreed

File: contracts/contracts/RequestManager.sol [hunk 3]
```diff
@@ -211,7 +211,7 @@ contract RequestManager is Ownable {
         );
         require(request.activeClaims == 0, "Active claims running");
 
-        request.depositReceiver = request.sender;
+        request.withdrawInfo = BeamerUtils.FillInfo(request.sender, bytes32(0));
```
Comment (palango) [inline contracts/contracts/RequestManager.sol:214]: Wouldn't it be enough to just set the filler here?

**Review by palango — APPROVED** [review]
LGTM

File: contracts/contracts/RequestManager.sol (hunk 3 above)
Comment (fredo) [reply contracts/contracts/RequestManager.sol:214]: totally right

**Pull request merged**